third_party/proton/csrc/lib/Profiler/RoctracerProfiler.cpp

```diff
@@ -0,0 +1,285 @@
+#include "Profiler/RoctracerProfiler.h"
+#include "Context/Context.h"
+#include "Data/Metric.h"
+#include "Driver/GPU/Hip.h"
+#include "Driver/GPU/Roctracer.h"
+
+#include <roctracer/roctracer_ext.h>
+#include <roctracer/roctracer_hip.h>
+#include <roctracer/roctracer_hsa.h>
+
+#include <cstdlib>
+#include <deque>
+#include <memory>
+#include <mutex>
+
+#include <unistd.h>
+
+namespace proton {
+
+namespace {
+
+// Local copy of public (and stable) HIP op kinds
+
+typedef enum {
+  HIP_OP_COPY_KIND_UNKNOWN_ = 0,
+  HIP_OP_COPY_KIND_DEVICE_TO_HOST_ = 0x11F3,
+  HIP_OP_COPY_KIND_HOST_TO_DEVICE_ = 0x11F4,
+  HIP_OP_COPY_KIND_DEVICE_TO_DEVICE_ = 0x11F5,
+  HIP_OP_COPY_KIND_DEVICE_TO_HOST_2D_ = 0x1201,
+  HIP_OP_COPY_KIND_HOST_TO_DEVICE_2D_ = 0x1202,
+  HIP_OP_COPY_KIND_DEVICE_TO_DEVICE_2D_ = 0x1203,
+  HIP_OP_COPY_KIND_FILL_BUFFER_ = 0x1207
+} hip_op_copy_kind_t_;
```

Comment:
Not used now. I think we need it in the future when categorizing memory operations. But can add when used. So dropped now.

Reply:
hip_op_copy_kind_t_ and hip_op_dispatch_kind_t_ are 2 parts of the same whole.  The roctracer_record_t.kind will contain one of those values to describe the async op.  The cupti reference was ignoring memcpy ops, so we are here as well.  Otherwise these would have been used.

Yes this is ugly on the roctracer end.